packages/discovery-react-components/src/components/StructuredQuery/components/AddRuleRowButton/AddRuleRowButton.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { FC, Dispatch, SetStateAction } from 'react';
+import { Button } from 'carbon-components-react';
+import Add16 from '@carbon/icons-react/lib/add/16';
+import { Messages } from 'components/StructuredQuery/messages';
+import { StructuredQuerySelection } from 'components/StructuredQuery/utils/structuredQueryInterfaces';
+
+export interface AddRuleRowButtonProps {
+  /**
+   * text to display for the Add rule button
+   */
+  addRuleRowText: Messages['addRuleRowText'];
+  /**
+   * id of the group for the rule row to render, or 'top-level' if the top-level rule group
+   */
+  groupId: number | 'top-level';
+  /**
+   * state that represents the current rules and selections for the structured query
+   */
+  structuredQuerySelection: StructuredQuerySelection;
+  /**
+   * used to set the structuredQuerySelection state
+   */
+  setStructuredQuerySelection: Dispatch<SetStateAction<StructuredQuerySelection>>;
+}
+
+const handleOnClick = (
```

Comment:
why did you move this outside of the component?

Reply:
I think this happened during splitting this out into its own component. Thanks for catching, updated!